api/src/utils/merge-version-data.ts

```diff
@@ -0,0 +1,186 @@
+import type { Item, SchemaOverview } from '@directus/types';
+import Joi from 'joi';
+import type { Alterations } from '../types/index.js';
+import { isObject } from '@directus/utils';
+
+const alterationSchema = Joi.object({
+	create: Joi.array().items(Joi.object().unknown()),
+	update: Joi.array().items(Joi.object().unknown()),
+	delete: Joi.array().items(Joi.string(), Joi.number()),
+});
+
+export function mergeVersionsRaw(item: Item, versionData: Partial<Item>[]) {
+	const result = { ...item };
```

Comment:
I believe this is intended to be a clone, but be aware that a top level `...` doesn't clone the whole nested structure. Any arrays or objects that are root properties are still by references instead of copied. We should probably use lodash' `cloneDeep` here to be safe.